email-validator.inc.php

```diff
@@ -52,7 +52,7 @@ function validateEMailAddress( $strEMailAddress )
 
 	function checkSyntax( &$strEMailAddress )
 	{
-		return preg_match( "/^[0-9a-z_]([-_\.]*[0-9a-z])*@[0-9a-z]([-\._]*[0-9a-z])*\\.[a-z]{2,3}$/i", $strEMailAddress  ) == 1 ;
```

Comment:
Email addresses with a '+' in them represent valid email addresses (even though they can be somewhat abused). I use them all the time, when I sign up for newsletters etc. to make sure, I know exactly who passed on my email address if I should receive spam. Also there are new top-level domains, that have up to 24! (insane) characters. 
`wget -qO - http://data.iana.org/TLD/tlds-alpha-by-domain.txt | tail -n+2 | less` shows all allowed top level domains.